### DISCUSSION TITLE
eloquent setrelation

### DISCUSSION THREAD

**Discussion by slushpuppy** [nuwave/lighthouse#2631] [Q&A] [answered]
hi, my root resolver returns an eloquent model array with hasOne relationship hydrated using the `setRelation` method, however when debugging the request, it appears graphql still issues a subquery call on that relationship, How do I prevent this 2nd subquery call from happening?

**Comment by stayallive** [comment, marked as answer]
I believe you should be able to remove the `@relation` directive to prevent that.
